cis/publisher.py

```diff
@@ -55,24 +56,28 @@ def _connect_aws(self):
             type='session',
             region='us-west-2'
         ).connect()
+        logger.debug('Connected to AWS using a boto_session.')
 
     def send(self):
         """
         :return: True or False based on exception.
         """
 
         if not self.boto_session:
+            logger.debug('No boto_session present in object. Connecting to AWS.')
             self._connect_aws()
 
         if self._validate_profile_data():
             result = self._invoke_validator(json.dumps(self._get_event_dict()).encode())
-            logger.info('Invocation result is {result}'.format(result=result))
+            logger.info('Validator invocation complete with result: {} for user_id: {}'.format(
+                result, self.profile_data.get('user_id'))
+            )
             return result
 
     def _get_event_dict(self):
-
-        encrypted_profile = str(base64.b64encode(self._prepare_profile_data()))
-        signature = self._generate_signature()
+        profile_data = self._prepare_profile_data()
+        encrypted_profile = str(base64.b64encode(profile_data))
+        signature = self._generate_signature(json.dumps(self.profile_data))
```

Comment:
should check if `_generate_signature` failed here, or in `_generate_signature`

Reply:
Signature generation is non-optional the way pykmssig works.  If a signature can not be generated it will raise.